zcash_client_sqlite/src/wallet/transparent/ephemeral.rs

```diff
@@ -118,12 +116,29 @@ pub(crate) fn get_ephemeral_ivk<P: consensus::Parameters>(
     params: &P,
     account_id: AccountId,
 ) -> Result<EphemeralIvk, SqliteClientError> {
-    Ok(get_account(conn, params, account_id)?
-        .ok_or(SqliteClientError::AccountUnknown)?
-        .ufvk()
+    let ufvk = conn
+        .query_row(
+            "SELECT ufvk FROM accounts WHERE id = :account_id",
+            named_params![":account_id": account_id.0],
```

Comment:
Why is this being altered to not use `get_account`? AFAICT none of the changes in this PR affect this code, and it should still work fine as it was before?

The refactored code is also subtly different. Previously this would return `Err(SqliteClientError::AccountUnknown)` for an unknown account; now it returns `Err(SqliteClientError::UnknownZip32Derivation)` (merging the two failure cases).

Reply:
This is a bit annoying: the call here to `get_account` now breaks when used in migration tests for migrations prior to the addition of the `has_spending_key` field; now that we retrieve the value of that field in `get_account`, the migration tests break. Instead of having to copy a stack of business logic into the migration tests, minimizing this to query for only what it needs seemed like the better approach; the code here is pretty trivial.